pymeasure/instruments/common_base.py

```diff
@@ -504,6 +505,7 @@ def measurement(get_command, docs, values=(), map_values=None,
         :param check_get_errors: Toggles checking errors after getting
         :param dynamic: Specify whether the property parameters are meant to be changed in
             instances or subclasses. See :meth:`control` for an usage example.
+        :param \\**kwargs: Keyword arguments for :meth:`values`.
```

Comment:
`setting()` will also need that addition I think.

Reply:
As far as I know, does setting **set** and not get, therefore it does not invoke `values`.

Reply:
Right. Should we then remove the `kwargs` from the signature of `setting`, or trigger a warning inside `setting` if kwargs is not None?

Reply:
I'm for removing it from the signature as we are not talking about a specific kwarg, but the general collection of all additionally used ones.

Reply:
Done.

Reply:
Thanks. Changelog entry "Unused **kwargs argument in Instrument.settings removed" or so)? ([Hyrum's law and all that](https://www.hyrumslaw.com/))

Reply:
I'll add an entry. Thanks for the link.

Reply:
Good, that we removed the "kwargs" call, as several tests failed, because the gave "get_process" to a setting etc.
Also the documentation failed with doctests...

Reply:
I see, that `setting` and `measurement` have different signatures (as they only require subsets of `control`).
If we keep `kwargs` in the signature, you can change more easily from one property creator type to the other, as superfluos kwargs do not stop it working.

So, instead of removing kwargs, we add a note, that they are dropped?

Reply:
> Good, that we removed the "kwargs" call, as several tests failed, because the gave "get_process" to a setting etc.

It's good that these things are uncovered!

> If we keep kwargs in the signature, you can change more easily from one property creator type to the other, as superfluos kwargs do not stop it working.

Is that desirable, though? If you change factory method, you should also adapt the argument to the appropriate ones. Removing **kwargs would highlight the ones that need changing, no?

Reply:
From a small "docstring and signature PR" to a large, multi file PR 🤦

Reply:
Well, such is life. 😂 you go where the journey leads you...
Also nicely demonstrates how small changes in the core can have amplified impact... 😁

Reply:
I'll do it. So stay tuned for more changes...